mp4parse_capi/src/lib.rs

```diff
@@ -1177,26 +1185,33 @@ fn mp4parse_get_track_video_info_safe(
 /// # Safety
 ///
 /// This function is unsafe because it dereferences both the parser and
-/// primary_item raw pointers passed into it. Callers should ensure the parser
-/// pointer points to a valid `Mp4parseAvifParser`, and that the primary_item
-/// pointer points to a valid `Mp4parseByteData`. If there was not a previous
+/// avif_image raw pointers passed into it. Callers should ensure the parser
+/// pointer points to a valid `Mp4parseAvifParser`, and that the avif_image
+/// pointer points to a valid `AvifImage`. If there was not a previous
 /// successful call to `mp4parse_avif_read()`, no guarantees are made as to
-/// the state of `primary_item`.
+/// the state of `avif_image`. If `avif_image.alpha_item` are set with a
+/// positive`length` and non-null `data`, then the `avif_image` contains an
+/// valid alpha channel data. Otherwise, the image is opaque.
 #[no_mangle]
-pub unsafe extern "C" fn mp4parse_avif_get_primary_item(
+pub unsafe extern "C" fn mp4parse_avif_get_image(
     parser: *mut Mp4parseAvifParser,
-    primary_item: *mut Mp4parseByteData,
+    avif_image: *mut AvifImage,
 ) -> Mp4parseStatus {
-    if parser.is_null() {
+    if parser.is_null() || avif_image.is_null() {
         return Mp4parseStatus::BadArg;
     }
 
     // Initialize fields to default values to ensure all fields are always valid.
-    *primary_item = Default::default();
+    *avif_image = Default::default();
+    let image_ref = &mut (*avif_image);
 
     let context = (*parser).context();
 
-    (*primary_item).set_data(context.primary_item());
+    image_ref.primary_item.set_data(context.primary_item());
+    if let Some(context_alpha_item) = context.alpha_item() {
+        image_ref.alpha_item.set_data(context_alpha_item);
+        image_ref.premultiplied_alpha = context.premultiplied_alpha;
+    }
```

Comment:
For my own curiosity, is there any reason you added the `image_ref` binding as opposed to using `avif_image` directly like so?
```rust
    (*avif_image).primary_item.set_data(context.primary_item());
    if let Some(context_alpha_item) = context.alpha_item() {
        (*avif_image).alpha_item.set_data(context_alpha_item);
        (*avif_image).premultiplied_alpha = context.premultiplied_alpha;
    }
```
This works perfectly fine, so I'm happy to merge it. It just threw me for a bit reading it, so I was interested in the rationale.

Reply:
I just worry the `*avif_image` would create a temporary variable cloned from `avif_image`. But I guess this won't happen in Rust. I wrote too much C++.

Reply:
Ah, that makes sense. I have trouble switching my mental model back and forth too, but my default is rust. Thanks for the explanation!